tests/unit/test_ast_utils.py

```diff
@@ -0,0 +1,19 @@
+#
+# Copyright (c) 2012-2024 Snowflake Computing Inc. All rights reserved.
+#
+
+from decimal import Decimal
+
+import pytest
+
+import snowflake.snowpark._internal.proto.ast_pb2 as proto
+from snowflake.snowpark._internal.ast_utils import build_expr_from_python_val
+
+
+@pytest.mark.parametrize(
+    "value", [0, 1, -1, 3.141, "1234567", "12345678.9876543212345676543456"]
+)
+def test_decimal_encoding(value):
+    v = Decimal(value)
+    expr = proto.Expr()
+    build_expr_from_python_val(expr, v)
```

Comment:
Optional: The test would be even more useful if it checked the value returned from `build_expr_from_python_val`.

Reply:
I started with this, but didn't want to add a dedicated decode routine. Thus this test just safeguards against the code not throwing exceptions.